Assigment3/EmployeeManagement.java

```diff
@@ -0,0 +1,307 @@
+import java.io.*;
+import java.util.*;
+
+/**
+ * EmployeeManagement class
+ * empList - stores the all data of employee from emp.csv
+ * empFile - BufferedReader instance used to read data from .csv
+ * latestIdAllocated - which initially set to last id allocated to employee
+ * location - represents file location initialized in constructor
+ * saveEmp - used to save data in file .csv
+ * changeStatus - it used to detect is there any changes made in data by add/delete/update operation
+ */
+class EmployeeManagement{
+    private HashMap<Integer,Employee> empList;
+    private BufferedReader empFile;
+    private int latestIdAllocated;
+    private File location;
+    private BufferedWriter saveEmp;
+    private boolean changeStatus; 
+
+    EmployeeManagement(){
+        changeStatus=false;
+        location=new File("./emp.csv");
+        getAllEmployeeData();
+        showAllEmployee();
+    }
+
+    /**Feaching all data from file and Storing in a empList DS */
+    public void getAllEmployeeData(){
+        this.empList=new HashMap<>();
+        this.latestIdAllocated=0;
+        try {
+            // connecting with file
+            FileReader fr=new FileReader(this.location);
+            // class instance
+            this.empFile=new BufferedReader(fr);
+
+            // started reading empdata
+            String empData=this.empFile.readLine(); 
+            while(empData!=null){
+                Employee tempEmp=getEmployeeByString(empData);
+                empList.put(tempEmp.getUid() , tempEmp);
+    
+                
+                this.latestIdAllocated=Math.max(this.latestIdAllocated,tempEmp.getUid());
+                // reading
+                empData=this.empFile.readLine();
+            }
+
+        } catch (Exception e) {
```

Comment:
Please use the exact exception which can be thrown from your program instead of using Exception class.